### PR TITLE
Additions for Fedora, Opensuse and general Linux

### PR DESCRIPTION
**Pull request by boredsquirrel**
I started by adding up-to-date version infos for Fedora. Then I added the instructions for Toolbox and Distrobox, toolbox being standard on Fedora Atomic, Distrobox on Opensuse microOS but also working on Fedora Atomic. 

These tools create containers of a Distro, so the QGis packages can be isolated from the host system.

GUI app entries are included in the explanations.

---

Then I added an instruction to add missing python dependencies using pip, under the main Linux chapter.

I also added a tutorial how to enforce XWayland, as Wayland shows a warning of missing features. This is opt-in but automated.

### CODE REVIEW

**Comment by boredsquirrel** [comment]
Question: This version number tables add unnecessary maintenance work. Just linking to a site having the version numbers would seem to make more sense.

Oh and I also added two ways to remove the Fedora COPR, as its deprecated and could cause conflicts

File: source/site/forusers/alldownloads.rst [hunk 1]
```diff
@@ -115,6 +115,17 @@ Below you find instructions to install per distribution. For most distro's
 there are instructions to install QGIS stable and instructions to install a
 cutting edge QGIS testing build (note the warning_).
 
+If Plugins report missing packages, you can install them with pip::
+
+   pip install PACKAGENAME
+
+QGis may report lacking features due to running as a Wayland session. To force X11 (XWayland) you can add ``env QT_QPA_PLATFORM=xcb`` to the ``Exec=`` line. If you installed the app on your main system, copy the entry to your user directory first, if you are using Toolbox/Distrobox or already have this entry, skip this step::
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst:122]:
```suggestion
QGIS may report lacking features due to running as a Wayland session. To force X11 (XWayland) you can add ``env QT_QPA_PLATFORM=xcb`` to the ``Exec=`` line. If you installed the app on your main system, copy the entry to your user directory first, if you are using Toolbox/Distrobox or already have this entry, skip this step::
```

Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
QGIS may report lacking features due to running as a Wayland session. To force X11 (XWayland) you can add ``env QT_QPA_PLATFORM=xcb`` to the ``Exec=`` line. If you installed the app on your main system, copy the entry to your user directory first, if you are using Toolbox/Distrobox or already have this entry, skip this step::
```

File: source/site/forusers/alldownloads.rst [hunk 2]
```diff
@@ -346,72 +357,50 @@ In case you would like to install QGIS Server, type::
...
-|               | 37          | 3.28         | 8.2          |
-+---------------+-------------+--------------+--------------+
-
-More information are available at https://copr.fedorainfracloud.org/coprs/dani/qgis/
-
-QGIS LTR (Long Term Release)
-............................
-
-Enable the repository::
+To install QGis on Fedora Atomic (Silverblue, Kinoite, Sericea etc.), it's best to create a Toolbox and install it there::
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
To install QGIS on Fedora Atomic (Silverblue, Kinoite, Sericea etc.), it's best to create a Toolbox and install it there::
```

File: source/site/forusers/alldownloads.rst [hunk 3]
```diff
@@ -346,72 +357,50 @@ In case you would like to install QGIS Server, type::
...
 |               |             | version      | version      |
 |               |             |              |              |
 +===============+=============+==============+==============+
-| Fedora        | 36          | 3.22         | 8.0          |
+| Fedora        | 38          | 3.28.11      | 8.2          |
+|               +-------------+--------------+--------------+
+|               | 39          | 3.32.2       | 8.3          |
 |               +-------------+--------------+--------------+
-|               | 37          | 3.22         | 8.2          |
+|               | Rawhide     | 3.34.0-3     | 8.3          |
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]: maybe 3.34.0, not 3.34.0-3?

File: source/site/forusers/alldownloads.rst [hunk 4]
```diff
@@ -346,72 +357,50 @@ In case you would like to install QGIS Server, type::
...
 +---------------+-------------+--------------+--------------+
 
-More information are available at https://copr.fedorainfracloud.org/coprs/dani/qgis-ltr/
-
+Updated Version info: `Qgis <https://packages.fedoraproject.org/pkgs/qgis/qgis/>` and `Grass <https://packages.fedoraproject.org/pkgs/grass/grass/>`
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
Updated Version info: `QGIS <https://packages.fedoraproject.org/pkgs/qgis/qgis/>` and `GRASS GIS <https://packages.fedoraproject.org/pkgs/grass/grass/>`
```

File: source/site/forusers/alldownloads.rst [hunk 5]
```diff
@@ -443,6 +432,20 @@ All packages include GRASS and Python support.
 All openSUSE Geo repositories can be found here:
 https://download.opensuse.org/repositories/Application:/Geo/
 
+To install QGis packages on OpenSuse:
+
+   sudo zypper in qgis qgis-plugin-grass
+
+To install QGis on OpenSuse microOS (now Aeon or Kalpa), use Distrobox:
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
To install QGIS packages on OpenSuse:

   sudo zypper in qgis qgis-plugin-grass

To install QGIS on OpenSuse microOS (now Aeon or Kalpa), use Distrobox:
```

**Review by agiudiceandrea — CHANGES_REQUESTED** [review]

File: source/site/forusers/alldownloads.rst (hunk 3 above)
Comment (agiudiceandrea) [reply source/site/forusers/alldownloads.rst]:
```suggestion
|               | Rawhide     | 3.34.0       | 8.3          |
```

File: source/site/forusers/alldownloads.rst [hunk 6]
```diff
@@ -115,6 +115,17 @@ Below you find instructions to install per distribution. For most distro's
 there are instructions to install QGIS stable and instructions to install a
 cutting edge QGIS testing build (note the warning_).
 
+If Plugins report missing packages, you can install them with pip::
+
+   pip install PACKAGENAME
+
+QGIS may report lacking features due to running as a Wayland session. To force X11 (XWayland) you can add ``env QT_QPA_PLATFORM=xcb`` to the ``Exec=`` line. If you installed the app on your main system, copy the entry to your user directory first, if you are using Toolbox/Distrobox or already have this entry, skip this step::
```
Comment (DelazJ) [inline source/site/forusers/alldownloads.rst]:
```suggestion
QGIS may report lacking features due to running as a Wayland session. To force X11 (XWayland) you can add ``env QT_QPA_PLATFORM=xcb`` to the ``Exec=`` line. If you installed the app on your main system, copy the entry to your user directory first. If you are using Toolbox/Distrobox or already have this entry, skip this step::
```
assuming I understand

File: source/site/forusers/alldownloads.rst [hunk 7]
```diff
@@ -115,6 +115,17 @@ Below you find instructions to install per distribution. For most distro's
...
+
+   sudo cp /usr/share/applications/org.qgis.qgis.desktop ~/.local/share/applications/
+
+To add the X11 environment variable automatically::
+
+   sed -i 's/^Exec=/Exec=env QT_QPA_PLATFORM=xcb /' ~/.local/share/applications/org.qgis.qgis.desktop
```
Comment (DelazJ) [inline source/site/forusers/alldownloads.rst]: What is the context of this? I have been using Linux distros for years and never had to consider these options* and I wonder if putting them here does not undermine the simplification effort we do with the "quickstart" section below. I honestly wouldn't start with this.

\* _TBH I barely understand what it is about and I am afraid a number of OSS sympathisers are as noob as myself_

Comment (boredsquirrel) [reply source/site/forusers/alldownloads.rst]: i just put it there as it is not distro specific. It is not easy to find this argument. For me on Wayland QGis always shows this "meerp I am in Wayland and have reduced features"

I dont know where the issuetracker for that is, it should for sure not be necessary anymore, but its just one command that adds the parameter automatically, so its not actually hard for users.

Comment (boredsquirrel) [reply source/site/forusers/alldownloads.rst]: an update: Researching, some people say QT5 had bad support for Wayland. I disagree as KDE works pretty great, but there currently is no list of features that are missing, or breakages.

Forcing this variable will make the app use XWayland, which will also be supported on future Fedora releases for example. Just not X11 native, out of the box.

I would just keep this tip here, as on some versions X11 is set with a variable (Ubuntu ?) and on some others it is not (Fedora, OpenSuse).

**Comment by boredsquirrel** [comment]
https://github.com/qgis/QGIS/issues/46031

Issue about the Wayland warning

File: source/site/forusers/alldownloads.rst (hunk 2 above)
Comment (agiudiceandrea) [reply source/site/forusers/alldownloads.rst]: @trytomakeyouprivate, it looks like you missed this suggested change.

**Comment by frankbroniewski** [comment]
Hello, whilst tackling https://github.com/qgis/QGIS-Website/issues/1224 I noticed the proposed changes in the documentation concerning Python package installation instructions.

Because I recently had to tackle Python package installations on OpenSuse, I learned about [PEP668](https://peps.python.org/pep-0668/) and the implications on installing packages via pip. Hence I allowed myself to propose the following PR to describe Python module installation on Linux Systems: https://github.com/qgis/QGIS-Website/pull/1225

In short, PEP668 lets Linux system distributions forbid to install Python packages system wide via pip in order to maintain system integrity. It is encouraged instead to install Python packages in a virtual environment, which can be done via [pipx](https://github.com/pypa/pipx), which should be widely available in the distribution's repository.

**Comment by boredsquirrel** [comment]
I am on Fedora Atomic and pip falls back to user installation automatically. `pip install --user` should do that.

Also cleaned up my additions a lot, should be way better now. Would be nice to have it merged or resolved soon.

File: source/site/forusers/alldownloads.rst (hunk 7 above)
Comment (boredsquirrel) [reply source/site/forusers/alldownloads.rst]: an update: the specific issue is about toolbars being dockable and possible to move around, make into standalone windows and dock them again etc. I heard on Wayland this is completely broken. The toolbar would be basically gone.

If you dont know what this is, your distro probably still uses X11. 

I agree that it is a bit much, but for now this is a hard requirement to make it run on Wayland.

**Comment by frankbroniewski** [comment]
No, unfortunately, `pip install --user` fails as well:

```
frank@schnipsel:~> pip install --user flask
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try
    zypper install python311-xyz, where xyz is the package
    you are trying to install.
    
[snip]
```

I tried OpenSuse Tumbleweed and Ubuntu 23.04. - on all of them `pip install` will fail with the above error message. Fedora does not yet implement PEP668, so that's why it is still working for you, but there's a [proposal](https://fedoraproject.org/wiki/Changes/PythonMarkExternallyManaged) to integrate this as well. Arch based distros do seem to implement this already as well, as a quick Google search revealed.

You can still circumvent using `pipx` or a `venv` by passing the `--break-system-packages` flag to pip, but this probably scares people - as intended.

**Comment by boredsquirrel** [comment]
hmm,would you mind adding the best simple solution, using a venv called "qgis" for example? Didnt know this was only working on Fedora

**Comment by frankbroniewski** [comment]
Sure, I already made a PR https://github.com/qgis/QGIS-Website/pull/1225 for that, that got already accepted 🙂

I am a noob when it comes to venv, I always avoided them as unnecessary complication - that bites me in the back now 😁

I can only quote the message that comes up when I try to install a package via pip ... this basically just repeats the steps in my PR - just leaving out the `python3.11 -m venv path/to/venv` part, since I have no practical experience with it.

> To install Python packages system-wide, try
>  zypper install python311-xyz, where xyz is the package
>  you are trying to install.
>    
>   If you wish to install a non-rpm packaged Python package,
>   create a virtual environment using python3.11 -m venv path/to/venv.
>    Then use path/to/venv/bin/python and path/to/venv/bin/pip.
>    
>   If you wish to install a non-rpm packaged Python application,
>   it may be easiest to use `pipx install xyz`, which will manage a
>   virtual environment for you. Install pipx via `zypper install python311-pipx` .

Theoretically, practically you just need to create a working directory for your project and
1. python -m venv ~/qgis
2. cd ~/qgis
3. source bin/activate
4. pip install PACKAGE
5. and use an editor with venv support

which seems quite a hassle to me. `pipx` works just like pip, it just does some magic behind the scenes - so I use that instead of virtual environments ...

**Comment by boredsquirrel** [comment]
okay, @frankbroniewski didnt know how to rebase on your version and reapply my diffs, so I just copy-pasted your python instructions into my branch.

Should be ready to merge now.

**Comment by frankbroniewski** [comment]
Hey, no worries, I struggle all the time with git when manual merging is required or something else out of the ordinary 😁
